spec/vcloud_director/generators/compute/instantiate_vapp_template_params_spec.rb

```diff
@@ -256,4 +256,135 @@ def self.common_cpu_assertions(xml)
+          conf = vapp_net_conf.xpath('./vcloud:NetworkConfig/vcloud:Configuration')
+          conf.xpath('./vcloud:FenceMode').text.must_equal 'bridged'
+          conf.xpath('./vcloud:ParentNetwork/@href').text.must_equal 'ENDPOINT/network/parent-network-id'
+          conf.xpath('./vcloud:RetainNetInfoAcrossDeployments').text.must_equal 'false'
+          conf.xpath('./vcloud:RouterInfo/vcloud:ExternalIp').text.must_equal '17.17.17.17'
+          subnet = conf.xpath('./vcloud:IpScopes/vcloud:IpScope')
+          subnet.xpath('./vcloud:IsEnabled').text.must_equal 'true'
+          subnet.xpath('./vcloud:IsInherited').text.must_equal 'false'
+          subnet.xpath('./vcloud:Gateway').text.must_equal '1.2.3.4'
+          subnet.xpath('./vcloud:Netmask').text.must_equal '255.255.255.0'
+          subnet.xpath('./vcloud:Dns1').text.must_equal '5.6.7.8'
+          subnet.xpath('./vcloud:Dns2').text.must_equal '9.10.11.12'
+          ip_range = subnet.xpath('./vcloud:IpRanges/vcloud:IpRange')
+          ip_range.xpath('./vcloud:StartAddress').text.must_equal '192.168.254.100'
+          ip_range.xpath('./vcloud:EndAddress').text.must_equal '192.168.254.199'
+        end
+      },
+      {
+        :case          => 'two vapp networks',
+        :vapp_networks => [
+          {
+            :name   => 'network1',
+            :subnet => [
+              {
+                :gateway => '1.1.1.1',
+                :netmask => '255.255.255.1'
+              }
+            ]
+          },
+          {
+            :name   => 'network2',
+            :subnet => [
+              {
+                :gateway => '2.2.2.2',
+                :netmask => '255.255.255.2'
+              }
+            ]
+          }
+        ],
+        :expect        => lambda do |vapp_net_conf|
+          common_vapp_network_config_assertions(vapp_net_conf)
+          vapp_net_conf.xpath('./vcloud:NetworkConfig').count.must_equal 2
+          vapp_net_conf.xpath('./vcloud:NetworkConfig[1]/@networkName').text.must_equal 'network1'
+          vapp_net_conf.xpath('./vcloud:NetworkConfig[2]/@networkName').text.must_equal 'network2'
+          conf1 = vapp_net_conf.xpath('./vcloud:NetworkConfig[1]/vcloud:Configuration')
+          subnet1 = conf1.xpath('./vcloud:IpScopes/vcloud:IpScope')
+          subnet1.xpath('./vcloud:Gateway').text.must_equal '1.1.1.1'
+          subnet1.xpath('./vcloud:Netmask').text.must_equal '255.255.255.1'
+          conf2 = vapp_net_conf.xpath('./vcloud:NetworkConfig[2]/vcloud:Configuration')
+          subnet2 = conf2.xpath('./vcloud:IpScopes/vcloud:IpScope')
+          subnet2.xpath('./vcloud:Gateway').text.must_equal '2.2.2.2'
+          subnet2.xpath('./vcloud:Netmask').text.must_equal '255.255.255.2'
+        end
+      }
+    ].each do |args|
+      it args[:case].to_s do
+        input = {
+          :name          => 'VAPP_NAME',
+          :vapp_networks => args[:vapp_networks]
+        }
+        input[:endpoint] = 'ENDPOINT'
```

Comment:
Any reason not to set `:endpoint` in L373?

Reply:
No, why?! 🙈